### PR TITLE
Use `find_by` to avoid error on offerable courses

### PR DESCRIPTION
**Pull request by cwrw**
## Context

When a provider attempts to make an offer on/view the offer tab for a course that is not open on apply and not exposed on find they are getting a 404. This is because the queries are using `find` which errors if we cannot find the `id` of the course in the `offerable_courses` collection.

## Changes proposed in this pull request

Change `find` to `find_by` to avoid an error. This now returns an empty collection instead, and the pages will load.
This means that we can now do the following:
1) Make an offer to a course that is not exposed in find
2) We cannot make on offer to a course that is not open on apply though, and we will still see an error
3) View the offers tab on an application that has been offered, regardless of the course being open on apply or exposed in find
4) Can edit an offer when it is not exposed in find
5) Cannot edit an offer when it is not open on apply

## Guidance to review

Let me know if I missed anything, we should tackle the open on apply issue separately

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by despo — APPROVED** [review]
This looks sensible, but shouldn't we also display a message or somehow indicate the status of the course on the offer tab to avoid confusion in case they attempt to take further action, e.g. change the offer if they are able to do so?

**Comment by cwrw** [comment]
@despo good question, I'm double checking around dropping the offer validation all together for open courses, will update shortly

**Comment by cwrw** [comment]
@despo I've now removed the offer validation to check a course is open on Apply, and verified with James around intended behaviour. This means we won't be raising an error if a provider attempts to change an offer if the course is not open on apply etc. Let me know your thoughts

**Comment by cwrw** [comment]
update: found some issues while testing, back to the drawing board

**Review by richardpattinson — APPROVED** [review]
looks pretty uncontroversial to me, happy to merge

**Comment by cwrw** [comment]
@despo @richardpattinson I've now removed the `open_on_apply` restriction from offerable courses. Could you please have another look?

File: app/queries/get_change_offer_options.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ def available_courses(provider:)
   def available_study_modes(course:)
     CourseOption
       .selectable
-      .where(course: offerable_courses.find(course.id))
+      .where(course: offerable_courses.find_by(id: course.id))
```
Comment (despo) [inline app/queries/get_change_offer_options.rb:24]: Now that the `open_on_apply` restriction has been removed, is this change needed - as it will enable to easily identify issues with non retrievable courses allocated to offers.

**Review by despo — APPROVED** [review]
A question about using `find_by` but other than that it looks good 👍🏼

File: app/queries/get_change_offer_options.rb (hunk 1 above)
Comment (cwrw) [reply app/queries/get_change_offer_options.rb:24]: Good question. I think I'd like to keep it for now as this will avoid the offer tab 404s we might get. If this proves to add some strange behaviour we can reinstate/revisit
